timeout_manager.go

```diff
@@ -32,24 +32,18 @@ func (tm *TimeoutManager) SchedulePacketTimeout(packet PRUDPPacketInterface) {
 
 // AcknowledgePacket marks a pending packet as acknowledged. It will be ignored at the next resend attempt
 func (tm *TimeoutManager) AcknowledgePacket(sequenceID uint16) {
-	if packet, ok := tm.packets.Get(sequenceID); ok {
-		// * Acknowledge the packet
-		tm.packets.Delete(sequenceID)
-
-		// * Update the RTT on the connection if the packet hasn't been resent
-		if packet.SendCount() <= tm.streamSettings.RTTRetransmit {
+	tm.packets.RunAndDelete(sequenceID, func(_ uint16, packet PRUDPPacketInterface) {
+		if packet.SendCount() >= tm.streamSettings.RTTRetransmit {
```

Comment:
We should maybe keep the previous comments on here

```suggestion
	// * Acknowledge the packet
	tm.packets.RunAndDelete(sequenceID, func(_ uint16, packet PRUDPPacketInterface) {
		// * Update the RTT on the connection if the packet hasn't been resent
		if packet.SendCount() >= tm.streamSettings.RTTRetransmit {
```